lib/matplotlib/stackplot.py

```diff
@@ -107,9 +111,22 @@ def stackplot(axes, x, *args,
         first_line = center - 0.5 * total
         stack += first_line
 
+    if hide_empty:
+        if any(k in kwargs for k in ['edgecolor', 'interpolate', 'where']):
+            raise ValueError('hide_empty is not compatible with edgecolor, '
+                             'interpolate or where')
+    else:
+        edgecolor = kwargs.pop('edgecolor', None)
```

Comment:
I generally like the idea of having a `hide_kwargs` dict as the `if ... else ...` calls I wrote are rather ugly. But I don't think it would work here because `where` and `edgecolor` are unique to every iteration of the loop so that won't work for them.

It will only work for `interpolate` but then I thought it was better to treat all 3 arguments the same.

Reply:
You can update / make a new dictionary under the same conditional in the loop.

Reply:
Yes that makes sense.

I think at this point though I favor the solution of allowing `where` to be a list (see below). Are you also fine with that?